src/System.Net.Security/tests/FunctionalTests/UnixGssFakeNegotiateStream.cs

```diff
@@ -0,0 +1,166 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Authentication;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Net.Security.Tests
+{
+    internal class UnixGssFakeNegotiateStream : NegotiateStream
```

Comment:
In the UnitTests folder we use the fake/mock/shim with another meaning which is why I've renamed the previous implementation of the simulated network stream as `VirtualNetworkStream`. From what I understand, this is using actual OS APIs to simulate a Kerberos server.

Would renaming this type to something like `UnixGssServerVirtualNegotiateStream` be ok?